src/uu/ls/src/ls.rs

```diff
@@ -3206,13 +3211,13 @@ fn display_item_name(
     more_info: String,
     out: &mut BufWriter<Stdout>,
     style_manager: &mut Option<StyleManager>,
-    current_column: usize,
+    current_column: LazyCell<usize, Box<dyn FnOnce() -> usize + '_>>,
```

Comment:
I think we can use the byte offset instead, because we add this byte if it could possibly wrap, so we only need a lower bound on adding it. That should be cheaper to compute than the `ansi_width` computation.

I can even show that GNU does that. I did this in a terminal that fit the filename easily, but because emojis consist of many bytes, GNU will add the byte:
```
touch 🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀.foo
env TERM=xterm  LS_COLORS="*.foo=0;31;42" TIME_STYLE=+T ls  --color=always 🦀
🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀🦀.foo
```
I checked this by piping into `bat -A`.

Reply:
What about colors? (i.e. when called with `--color=auto`. That's the other thing that `ansi_width` removes...

Reply:
You should be able to just add the original byte size to some counter from before the color was added, right?

Reply:
Huh right, the color specifiers are just added in a few places. And then we could get rid of `ansi_width`.

I don't want to do this as part of this PR though, I think I'd need to investigate more deeply and this PR is meant to be a no-op in terms of functionality. Filed #7804.

Do you want me to drop the commit that adds this `LazyCell` thing from this PR? Either way is fine by me.